src/components/base/BaseButton.vue

```diff
@@ -3,18 +3,20 @@ import { computed } from "vue";
 
 const props = withDefaults(
   defineProps<{
-    buttonType:
-    | "primary"
-    | "primary-rev"
-    | "secondary"
-    | "tertiary"
-    | "tertiary-rev"
-    | "breadcrumb"
-    | "breadcrumb-rev";
-    buttonSize: "small" | "regular" | "medium" | "large" | "slim";
-    to: string;
+    buttonKind?: "button" | "submit";
+    buttonType?:
+      | "primary"
+      | "primary-rev"
+      | "secondary"
+      | "tertiary"
+      | "tertiary-rev"
+      | "breadcrumb"
+      | "breadcrumb-rev";
+    buttonSize?: "small" | "regular" | "medium" | "large" | "slim";
+    to?: string;
```

Comment:
Have you tried to set this up as the same type that router-link accepts? 🤔
This way we could use named routes which is a good practice to make our code resilient to path changes 👀

Reply:
Should be to hard. This was on the list of things to do anyhow and I believed I asked you about this possibility couple days back. On it 😸